core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java

```diff
@@ -0,0 +1,506 @@
+    return putValue(key, value);
+  }
+
+  @Override
+  public boolean setDefaultRaw(String key, byte[] defaultValue) {
+    return setDefaultValue(key, defaultValue);
+  }
+
+  @Override
+  public boolean putRaw(String key, ByteBuffer value, int len) {
+    byte[] buf = new byte[len];
+    value.get(buf, 0, len);
+    return putRaw(key, buf);
+  }
+
+  @Override
+  public byte[] getRaw(String key) throws TableKeyNotDefinedException {
+    return (byte[]) getValue(key);
+  }
+
+  @Override
+  public byte[] getRaw(String key, byte[] defaultValue) {
+    return (byte[]) getValue(key, defaultValue);
+  }
+
+  @Override
+  public void addTableListener(ITableListener listener) {
+    listeners.computeIfAbsent("", k -> new ArrayList<>()).add(listener);
+  }
+
+  @Override
+  public void addTableListener(ITableListener listener, boolean immediateNotify) {
+    if (immediateNotify) {
+      for (String key : getKeys()) {
+        listener.valueChanged(this, key, getValue(key), false);
+      }
+    }
+    addTableListener(listener);
+  }
+
+  @Override
+  public void addTableListenerEx(ITableListener listener, int flags) {
+    for (Map.Entry<String, Object> e : entries.entrySet()) {
+      listener.valueChangedEx(this, e.getKey(), e.getValue(), flags);
+    }
+  }
+
+  @Override
+  public void addTableListener(String key, ITableListener listener, boolean immediateNotify) {
+    if (immediateNotify) {
+      listener.valueChanged(this, key, getValue(key), false);
+    }
+    listeners.computeIfAbsent(key, k -> new ArrayList<>()).add(listener);
+  }
+
+  @Override
+  public void addTableListenerEx(String key, ITableListener listener, int flags) {
+    listener.valueChangedEx(this, key, getValue(key), flags);
+  }
+
+  @Override
+  public void addSubTableListener(ITableListener listener) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void addSubTableListener(ITableListener listener, boolean localNotify) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void removeTableListener(ITableListener listener) {
+    listeners.values().forEach(l -> l.remove(listener));
+  }
+
+  @Override
+  public boolean putInt(String key, int value) {
+    return putNumber(key, value);
+  }
+
+  @Override
+  public int getInt(String key) throws TableKeyNotDefinedException {
+    return (int) getNumber(key);
+  }
+
+  @Override
+  public int getInt(String key, int defaultValue) throws TableKeyNotDefinedException {
+    return (int) getValue(key, defaultValue);
+  }
+
+  @Override
+  public boolean putDouble(String key, double value) {
+    return putNumber(key, value);
+  }
+
+  @Override
+  public double getDouble(String key) throws TableKeyNotDefinedException {
+    return getNumber(key);
+  }
+
+  @Override
+  public double getDouble(String key, double defaultValue) {
+    return getNumber(key, defaultValue);
+  }
+
+  @Override
+  public String toString() {
+    return "MockNetworkTable: " + name;
```

Comment:
This whole class really should be provided by network tables as a testing library. @PeterJohnson What do you think about that?

@SamCarlberg Keep it here though for this PR.